### PR TITLE
ramips：Fixed the problem of MAC address and restart failure of hilink hlk-7628n.

### PR DESCRIPTION
**Pull request by liuyuf78fk**
1 ramips:Fix Ethernet random MAC address for HILINK HLK-7628N
2 ramips: Fixes the problem of software reboot failure in HILINK HLK-7628N

### CODE REVIEW

File: target/linux/ramips/dts/mt7628an_hilink_hlk-7628n.dts [hunk 1]
```diff
@@ -29,13 +29,6 @@
 	};
 };
 
-&state_default {
-	gpio {
-		groups = "i2c";
-		function = "gpio";
-	};
-};
-
```
Comment (adschm) [inline target/linux/ramips/dts/mt7628an_hilink_hlk-7628n.dts:38]: Why is this removed?

Comment (liuyuf78fk) [reply target/linux/ramips/dts/mt7628an_hilink_hlk-7628n.dts:38]: On HILINK-7628N, this is useless code.

File: target/linux/ramips/mt76x8/base-files/etc/board.d/02_network [hunk 2]
```diff
@@ -168,6 +168,7 @@ ramips_setup_macs()
 		lan_mac=$wan_mac
 		;;
 	cudy,wr1000|\
+	hilink,hlk-7628n|\
```
[the remark this answers is not in the document]
Comment (liuyuf78fk) [reply target/linux/ramips/mt76x8/base-files/etc/board.d/02_network:171]: I verified it. 
kernel log:
[    0.000000] Linux version 5.4.123 (liuyu@ubuntu-HARD) (gcc version 8.4.0 (OpenWrt GCC 8.4.0 r16859-2cd1a10829)) #0 Thu Jun 3 03:38:59 2021
[    0.000000] Board has DDR2
[    0.000000] Analog PMU set to hw control
[    0.000000] Digital PMU set to hw control
[    0.000000] SoC Type: MediaTek MT7628AN ver:1 eco:2
[    0.000000] printk: bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 00019655 (MIPS 24KEc)
[    0.000000] MIPS: machine is HILINK HLK-7628N
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Primary instruction cache 64kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, PIPT, no aliases, linesize 32 bytes
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] On node 0 totalpages: 32768
[    0.000000]   Normal zone: 288 pages used for memmap
[    0.000000]   Normal zone: 0 pages reserved
[    0.000000]   Normal zone: 32768 pages, LIFO batch:7
[    0.000000] pcpu-alloc: s0 r0 d32768 u32768 alloc=1*32768
[    0.000000] pcpu-alloc: [0] 0 
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 32480
[    0.000000] Kernel command line: console=ttyS0,57600 rootfstype=squashfs,jffs2
[    0.000000] Dentry cache hash table entries: 16384 (order: 4, 65536 bytes, linear)
[    0.000000] Inode-cache hash table entries: 8192 (order: 3, 32768 bytes, linear)
[    0.000000] Writing ErrCtl register=0003ff66
[    0.000000] Readback ErrCtl register=0003ff66
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory: 122304K/131072K available (4745K kernel code, 201K rwdata, 1048K rodata, 1228K init, 197K bss, 8768K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] NR_IRQS: 256
[    0.000000] intc: using register map from devicetree
[    0.000000] random: get_random_bytes called from start_kernel+0x358/0x54c with crng_init=0
[    0.000000] CPU Clock: 580MHz
[    0.000000] timer_probe: no matching timers found
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 6590553264 ns
[    0.000010] sched_clock: 32 bits at 290MHz, resolution 3ns, wraps every 7405115902ns
[    0.015389] Calibrating delay loop... 385.84 BogoMIPS (lpj=1929216)
[    0.087592] pid_max: default: 32768 minimum: 301
[    0.096914] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.111246] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.132726] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.152113] futex hash table entries: 256 (order: -1, 3072 bytes, linear)
[    0.165615] pinctrl core: initialized pinctrl subsystem
[    0.177025] NET: Registered protocol family 16
[    0.219208] workqueue: max_active 576 requested for napi_workq is out of range, clamping between 1 and 512
[    0.242015] clocksource: Switched to clocksource MIPS
[    0.253371] NET: Registered protocol family 2
[    0.262120] IP idents hash table entries: 2048 (order: 2, 16384 bytes, linear)
[    0.277033] tcp_listen_portaddr_hash hash table entries: 512 (order: 0, 4096 bytes, linear)
[    0.293517] TCP established hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.308589] TCP bind hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.322517] TCP: Hash tables configured (established 1024 bind 1024)
[    0.335216] UDP hash table entries: 256 (order: 0, 4096 bytes, linear)
[    0.348069] UDP-Lite hash table entries: 256 (order: 0, 4096 bytes, linear)
[    0.362170] NET: Registered protocol family 1
[    0.370667] PCI: CLS 0 bytes, default 32
[    0.383293] workingset: timestamp_bits=14 max_order=15 bucket_order=1
[    0.404673] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.416128] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.453309] mt7621_gpio 10000600.gpio: registering 32 gpios
[    0.464566] mt7621_gpio 10000600.gpio: registering 32 gpios
[    0.475799] mt7621_gpio 10000600.gpio: registering 32 gpios
[    0.487036] Serial: 8250/16550 driver, 3 ports, IRQ sharing disabled
[    0.500728] printk: console [ttyS0] disabled
[    0.509168] 10000c00.uartlite: ttyS0 at MMIO 0x10000c00 (irq = 28, base_baud = 2500000) is a 16550A
[    0.526972] printk: console [ttyS0] enabled
[    0.543462] printk: bootconsole [early0] disabled
[    0.562726] 10000d00.uart1: ttyS1 at MMIO 0x10000d00 (irq = 29, base_baud = 2500000) is a 16550A
[    0.581594] spi-mt7621 10000b00.spi: sys_freq: 193333333
[    0.608914] spi-nor spi0.0: w25q256 (32768 Kbytes)
[    0.618513] 4 fixed-partitions partitions found on MTD device spi0.0
[    0.631106] Creating 4 MTD partitions on "spi0.0":
[    0.640607] 0x000000000000-0x000000030000 : "u-boot"
[    0.651528] 0x000000030000-0x000000040000 : "u-boot-env"
[    0.663127] 0x000000040000-0x000000050000 : "factory"
[    0.674267] 0x000000050000-0x000002000000 : "firmware"
[    0.688878] 2 uimage-fw partitions found on MTD device firmware
[    0.700674] Creating 2 MTD partitions on "firmware":
[    0.710534] 0x000000000000-0x0000001e80e9 : "kernel"
[    0.721437] 0x0000001e80e9-0x000001fb0000 : "rootfs"
[    0.732380] mtd: device 5 (rootfs) set to be root filesystem
[    0.745388] 1 squashfs-split partitions found on MTD device rootfs
[    0.757710] 0x0000004e0000-0x000001fb0000 : "rootfs_data"
[    0.770294] libphy: Fixed MDIO Bus: probed
[    0.789810] rt3050-esw 10110000.esw: link changed 0x00
[    0.801975] mtk_soc_eth 10100000.ethernet eth0: mediatek frame engine at 0xb0100000, irq 5
[    0.820385] NET: Registered protocol family 10
[    0.833705] Segment Routing with IPv6
[    0.841100] NET: Registered protocol family 17
[    0.850048] 8021q: 802.1Q VLAN Support v1.8
[    0.868596] VFS: Mounted root (squashfs filesystem) readonly on device 31:5.
[    0.889632] Freeing unused kernel memory: 1228K
[    0.898640] This architecture does not have kernel memory protection.
[    0.911396] Run /sbin/init as init process
[    2.021535] init: Console is alive
[    2.028688] init: - watchdog -
[    2.502472] random: fast init done
[    3.188026] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    3.374136] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    3.399044] init: - preinit -
[    4.985194] random: jshn: uninitialized urandom read (4 bytes read)
[    5.085592] random: jshn: uninitialized urandom read (4 bytes read)
[    5.375583] random: jshn: uninitialized urandom read (4 bytes read)
[    8.979725] jffs2: notice: (398) jffs2_build_xattr_subsystem: complete building xattr subsystem, 6 of xdatum (0 unchecked, 1 orphan) and 7 of xref (1 dead, 0 orphan) found.
[    9.012487] mount_root: switching to jffs2 overlay
[    9.027783] overlayfs: upper fs does not support tmpfile.
[    9.044493] urandom-seed: Seed file not found (/etc/urandom.seed)
[    9.165740] procd: - early -
[    9.171559] procd: - watchdog -
[    9.821221] procd: - watchdog -
[    9.828190] procd: - ubus -
[    9.985869] urandom_read: 2 callbacks suppressed
[    9.985879] random: ubusd: uninitialized urandom read (4 bytes read)
[   10.013881] random: ubusd: uninitialized urandom read (4 bytes read)
[   10.027114] random: ubusd: uninitialized urandom read (4 bytes read)
[   10.043925] procd: - init -
[   11.108837] kmodloader: loading kernel modules from /etc/modules.d/*
[   11.318732] Loading modules backported from Linux version v5.10.34-0-g0aa66717f684
[   11.333825] Backport generated by backports.git v5.10.34-1-0-g7b5533e1
[   11.432946] xt_time: kernel timezone is -0000
[   11.565674] mt76_wmac 10300000.wmac: ASIC revision: 76280001
[   11.658883] urngd: v1.0.2 started.
[   11.867160] random: crng init done
[   11.873943] random: 2 urandom warning(s) missed due to ratelimiting
[   12.604211] mt76_wmac 10300000.wmac: Firmware Version: 20151201
[   12.616018] mt76_wmac 10300000.wmac: Build Time: 20151201183641
[   12.672033] mt76_wmac 10300000.wmac: firmware init done
[   12.852293] ieee80211 phy0: Selected rate control algorithm 'minstrel_ht'
[   12.970536] PPP generic driver version 2.4.2
[   13.003063] NET: Registered protocol family 24
[   13.044637] kmodloader: done loading kernel modules from /etc/modules.d/*
[   21.471715] rt3050-esw 10110000.esw: link changed 0x00
[   25.074525] rt3050-esw 10110000.esw: link changed 0x01
[   28.927356] br-lan: port 1(eth0.1) entered blocking state
[   28.938148] br-lan: port 1(eth0.1) entered disabled state
[   28.949442] device eth0.1 entered promiscuous mode
[   28.958992] device eth0 entered promiscuous mode
[   29.043400] br-lan: port 1(eth0.1) entered blocking state
[   29.054168] br-lan: port 1(eth0.1) entered forwarding state
[   29.946149] IPv6: ADDRCONF(NETDEV_CHANGE): br-lan: link becomes ready
[   92.438175] br-lan: port 2(wlan0) entered blocking state
[   92.448797] br-lan: port 2(wlan0) entered disabled state
[   92.459784] device wlan0 entered promiscuous mode
[   92.490069] IPv6: ADDRCONF(NETDEV_CHANGE): wlan0: link becomes ready
[   92.503012] br-lan: port 2(wlan0) entered blocking state
[   92.513588] br-lan: port 2(wlan0) entered forwarding state

root@OpenWrt:/# ifconfig
br-lan    Link encap:Ethernet  HWaddr 0C:CF:89:2A:E9:8B  
          inet addr:192.168.1.1  Bcast:192.168.1.255  Mask:255.255.255.0
          inet6 addr: fd19:8b73:913e:4::1/62 Scope:Global
          inet6 addr: fe80::ecf:89ff:fe2a:e98b/64 Scope:Link
          inet6 addr: fd16:b370:1e76::1/60 Scope:Global
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:3074 errors:0 dropped:0 overruns:0 frame:0
          TX packets:1741 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:429876 (419.8 KiB)  TX bytes:991047 (967.8 KiB)

eth0      Link encap:Ethernet  HWaddr 0C:CF:89:2A:E9:8B  
          inet6 addr: fe80::ecf:89ff:fe2a:e98b/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:3061 errors:0 dropped:1 overruns:0 frame:0
          TX packets:3151 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:1143940 (1.0 MiB)  TX bytes:493174 (481.6 KiB)
          Interrupt:5 

eth0.1    Link encap:Ethernet  HWaddr 0C:CF:89:2A:E9:8B  
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:91 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:0 (0.0 B)  TX bytes:33974 (33.1 KiB)

eth0.2    Link encap:Ethernet  HWaddr 0C:CF:89:2A:E9:8B  
          inet addr:172.16.8.10  Bcast:172.16.255.255  Mask:255.255.0.0
          inet6 addr: fd19:8b73:913e:0:ecf:89ff:fe2a:e98b/64 Scope:Global
          inet6 addr: fd4f:d4fd:8bda:0:ecf:89ff:fe2a:e98b/64 Scope:Global
          inet6 addr: fe80::ecf:89ff:fe2a:e98b/64 Scope:Link
          inet6 addr: fd19:8b73:913e::26e/128 Scope:Global
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:2758 errors:0 dropped:36 overruns:0 frame:0
          TX packets:3053 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:1075645 (1.0 MiB)  TX bytes:445798 (435.3 KiB)

lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:44 errors:0 dropped:0 overruns:0 frame:0
          TX packets:44 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:4879 (4.7 KiB)  TX bytes:4879 (4.7 KiB)

wlan0     Link encap:Ethernet  HWaddr 0C:CF:89:2A:E9:8A  
          inet6 addr: fe80::ecf:89ff:fe2a:e98a/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:3091 errors:0 dropped:0 overruns:0 frame:0
          TX packets:2009 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:474080 (462.9 KiB)  TX bytes:1043831 (1019.3 KiB)

root@OpenWrt:/#

[the remark this answers is not in the document]
Comment (liuyuf78fk) [reply target/linux/ramips/mt76x8/base-files/etc/board.d/02_network:171]: > Have you also verified the label MAC address?
> 
> Please add an overview like discussed here to your commit message so the OEM MAC address configuration is documented for future developers: https://openwrt.org/docs/guide-developer/device-support-policies#mac_addresses
I have modified the  commit message 。

File: target/linux/ramips/dts/mt7628an_hilink_hlk-7628n.dts (hunk 1 above)
Comment (adschm) [reply target/linux/ramips/dts/mt7628an_hilink_hlk-7628n.dts:38]: "Why is it green?" - "Because somebody painted it that way ..." :-(

Please tell me why you think this is useless... No, don't tell me. Put it in the commit message of your second commit!

Comment (adschm) [reply target/linux/ramips/dts/mt7628an_hilink_hlk-7628n.dts:38]: Looks like this shouldn't be removed but replaced by the correct values...

**Comment by adschm** [comment]
I've picked the MAC address-related part (with fixes) for now.

**Comment by liuyuf78fk** [comment]
Two submission, solved two problems, 1 MAC address random problem, 2 restart failure problem.

**Comment by adschm** [comment]
Sorry, I think we did too many unrelated changes in here. Please submit your remaining changes in a new PR, otherwise I don't think anybody will find them in here.
